src/include/sof/audio/module_adapter/module/module_interface.h

```diff
@@ -51,7 +51,8 @@ struct input_stream_buffer {
 	void __sparse_cache *data; /* data stream buffer */
 	uint32_t size; /* size of data in the buffer */
 	uint32_t consumed; /* number of bytes consumed by the module */
-
+	/* comp_buffer id, add to identify buffer of smart_amp*/
+	uint32_t id;
```

Comment:
It is strange that we have an ID field in the structure, but the id is something else's ID. I think it could be documented as input_stream_buffer ID, and it is assigned from comp_buffer, or you can rename it to comp_buff_id.

you can also add queue id, like https://github.com/thesofproject/sof/pull/7310